server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -3899,6 +3914,16 @@ private UserVm createVirtualMachine(DataCenter zone, ServiceOffering serviceOffe
         return vm;
     }
 
+    @Override
+    public Boolean checkIfDynamicScalingCanBeEnabled(ServiceOffering offering, VirtualMachineTemplate template, Long zoneId) {
```

Comment:
Can this method also accept a VM as a parameter and check if the VM is scalable as well?

Reply:
VM is separated because first VM deployment will not have VM flag already saved. But I felt putting a null check would simplify the code a bit. Made those changes.